common/sgx/eeid.c

```diff
@@ -356,19 +356,21 @@ oe_result_t oe_remeasure_memory_pages(
 
     if (with_eeid_pages)
     {
+        char* eeid_bytes = (char*)eeid;
         size_t num_bytes = oe_eeid_byte_size(eeid);
         size_t num_pages =
             num_bytes / OE_PAGE_SIZE + ((num_bytes % OE_PAGE_SIZE) ? 1 : 0);
 
-        oe_page_t* pages = malloc(sizeof(oe_page_t) * num_pages);
-        memset(pages, 0, sizeof(oe_page_t) * num_pages);
-        memcpy(pages->data, eeid, num_bytes);
-
+        oe_page_t* page = malloc(sizeof(oe_page_t) * num_pages);
```

Comment:
Even though `num_pages` number of pages are allocated, only the first page is used.

Reply:
Yes, I re-use the same bit of memory (it's reset to 0 in the loop). I don't think it's necessary to allocate a new page for each iteration, is it?

Reply:
Sorry, misunderstood you - yes, that's far too much memory :-)

Reply:
(fixed)